api/v1beta1/types.go

```diff
@@ -318,16 +318,45 @@ type Image struct {
 	ID *string `json:"id,omitempty"`
 
 	// SharedGallery specifies an image to use from an Azure Shared Image Gallery
+	// Deprecated: use ComputeGallery instead.
 	// +optional
 	SharedGallery *AzureSharedGalleryImage `json:"sharedGallery,omitempty"`
 
 	// Marketplace specifies an image to use from the Azure Marketplace
 	// +optional
 	Marketplace *AzureMarketplaceImage `json:"marketplace,omitempty"`
+
+	// ComputeGallery specifies an image to use from the Azure Compute Gallery
+	// +optional
+	ComputeGallery *AzureComputeGalleryImage `json:"computeGallery,omitempty"`
 }
 
-// AzureMarketplaceImage defines an image in the Azure Marketplace to use for VM creation.
-type AzureMarketplaceImage struct {
+// AzureComputeGalleryImage defines an image in the Azure Compute Gallery to use for VM creation.
+type AzureComputeGalleryImage struct {
+	// Gallery specifies the name of the compute image gallery that contains the image
+	// +kubebuilder:validation:MinLength=1
+	Gallery string `json:"gallery"`
+	// Name is the name of the image
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+	// Version specifies the version of the marketplace image. The allowed formats
+	// are Major.Minor.Build or 'latest'. Major, Minor, and Build are decimal numbers.
+	// Specify 'latest' to use the latest version of an image available at deploy time.
+	// Even if you use 'latest', the VM image will not automatically update after deploy
+	// time even if a new version becomes available.
+	// +kubebuilder:validation:MinLength=1
+	Version string `json:"version"`
+	// SubscriptionID is the identifier of the subscription that contains the shared image gallery.
+	// +optional
+	SubscriptionID *string `json:"subscriptionID,omitempty"`
+	// ResourceGroup specifies the resource group containing the shared image gallery.
+	// +optional
+	ResourceGroup *string `json:"resourceGroup,omitempty"`
+	// Plan contains plan information.
+	Plan *ImagePlan `json:"plan,omitempty"`
```

Comment:
should this be marked as optional?

Reply:
Hmm, it seems adding it has no effect on the manifests. But I'll add it for consistency.